Index: mail/mail-callbacks.h
===================================================================
RCS file: /cvs/gnome/evolution/mail/mail-callbacks.h,v
retrieving revision 1.52
diff -u -u -r1.52 mail-callbacks.h
--- mail/mail-callbacks.h	24 Jan 2003 04:59:35 -0000	1.52
+++ mail/mail-callbacks.h	17 Jul 2003 06:38:24 -0000
@@ -129,8 +129,6 @@
 void composer_send_cb      (EMsgComposer *composer, gpointer data);
 void composer_save_draft_cb (EMsgComposer *composer, int quit, gpointer data);
 
-void forward_messages	   (CamelFolder *folder, GPtrArray *uids, gboolean inline);
-
 /* CamelStore callbacks */
 void folder_created (CamelStore *store, const char *prefix, CamelFolderInfo *fi);
 void folder_deleted (CamelStore *store, CamelFolderInfo *fi);
