--- src/frontc/cabs2cil.ml.old	2006-05-20 23:14:15.000000000 -0500
+++ src/frontc/cabs2cil.ml	2006-10-24 19:07:26.000000000 -0500
@@ -148,6 +148,11 @@
   if !msvcMode then n = "va_list"  || n = "__ccured_va_list" 
   else n = "__builtin_va_alist_t"
 
+let isVariadicListType t =
+  match unrollType t with
+  | TBuiltin_va_list _ -> true
+  | _ -> false
+
 (* Weimer
  * multi-character character constants
  * In MSCV, this code works:
@@ -4191,6 +4196,12 @@
   | (Eq|Ne) when isPointerType t2 && isZero e1 -> 
       pointerComparison (mkCastT zero !upointType t2) t2 e2 t2
 
+  | (Eq|Ne) when isVariadicListType t1 && isZero e2 -> 
+      ignore (warnOpt "Comparison of va_list and zero");
+      pointerComparison e1 t1 (mkCastT zero !upointType t1) t1
+  | (Eq|Ne) when isVariadicListType t2 && isZero e1 -> 
+      ignore (warnOpt "Comparison of zero and va_list");
+      pointerComparison (mkCastT zero !upointType t2) t2 e2 t2
 
   | (Eq|Ne|Le|Lt|Ge|Gt) when isPointerType t1 && isArithmeticType t2 ->
       ignore (warnOpt "Comparison of pointer and non-pointer");
